### PR TITLE
Refactor: Use `Object.prototype` directly

### PR DESCRIPTION
**Pull request by fisker**
use `Object.prototype` directly, instead of `Object.getPrototypeOf({})`

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,10 +1,8 @@
 'use strict';
+const {prototype, getPrototypeOf} = Object;
+const {toString} = prototype;
+const prototypes = new Set([null, prototype]);
```
Comment (sindresorhus) [inline index.js]: Using a Set for just two elements is wasteful. The current check is both faster and simpler.

File: index.js [hunk 2]
```diff
@@ -1,10 +1,8 @@
 'use strict';
+const {prototype, getPrototypeOf} = Object;
+const {toString} = prototype;
```
Comment (sindresorhus) [inline index.js]: I don't really see the point of destructuring here. V8 is smart enough to optimize property access away, so this is just verbosity for no benefit.

**Comment by fisker** [comment]
@sindresorhus restored other changes, only replace `Object.getPrototypeOf({})` with `Object.prototype`

**Comment by fisker** [comment]
@sindresorhus Can you make a release?

**Comment by sindresorhus** [comment]
@fisker Why? Does it fix anything? I don't usually do new releases for refactorings and cosmetic changes.

**Comment by fisker** [comment]
@sindresorhus It's faster, no?

**Comment by sindresorhus** [comment]
Do you have any benchmark to prove that? And even if it's faster in a micro-benchmark, it most likely has no impact at all on real-world code. V8 is pretty good at optimizing stuff.

**Comment by fisker** [comment]
```js
const oldOne = value => {
	if (Object.prototype.toString.call(value) !== '[object Object]') {
		return false;
	}

	const prototype = Object.getPrototypeOf(value);
	return prototype === null || prototype === Object.getPrototypeOf({});
}

const newOne = value => {
	if (Object.prototype.toString.call(value) !== '[object Object]') {
		return false;
	}

	const prototype = Object.getPrototypeOf(value);
	return prototype === null || prototype === Object.prototype;
}

console.time('old')
for (let i = 0; i < 1e8; i++) {
  oldOne('')
  oldOne(Object.create(null))
  oldOne({})
}
console.timeEnd('old')

console.time('new')
for (let i = 0; i < 1e8; i++) {
  newOne('')
  newOne(Object.create(null))
  newOne({})
}
console.timeEnd('new')
```

About 20% faster

```text
$ nvm use 12.13.1
Now using node v12.13.1 (64-bit)

$ node 1.js
old: 9057.424ms
new: 7586.042ms

$ nvm use 10.17.0
Now using node v10.17.0 (64-bit)

$ node 1.js
old: 10626.064ms
new: 8113.567ms
```

**Comment by fisker** [comment]
If only test with this object `{}`, 

```text
old: 4453.476ms
new: 2380.431ms
```

@sindresorhus

**Comment by sindresorhus** [comment]
I'll do a new release so we don't have to waste more time on this conversation (I mean that in the kindest way possible. I've had similar conversations like this all too many times), but I still strongly disagree it has any practical benefit. You're measuring a simple function in isolation. That is no proof how it will perform in real code. And you would have to run it thousands of times to even have a slight impact.

**Comment by fisker** [comment]
Thank you very much.

**Pull request merged**